### ISSUE TITLE
[FEATURE REQ] Determine if EventHubProducerAsyncClient can send data to Event Hub without sending data (for healthcheck)

### ISSUE THREAD

**Issue by scootafew** [Azure/azure-sdk-for-java#22803]
**Is your feature request related to a problem? Please describe.**
I'm trying to write a healthcheck for an `EventHubProducerAsyncClient` (from azure-messaging-eventhubs), but I have not been able to find way to achieve this through the public API. As far as I can tell, the only way to determine if sending data to the Event Hub will be possible given the configuration is to actually send a message.

**Describe the solution you'd like**
I want to write a healthcheck that will at least fail if:
- The Event Hub or Event Hub Namespace does not exist or cannot be reached
- The Event Hub exists but is Disabled
- The connection string provided does not have the `Send` permission

Or will pass otherwise.

There is a method in `EventHubProducerAsyncClient` which as far as I can tell will initialise a send link without sending any data:

https://github.com/Azure/azure-sdk-for-java/blob/1023b0c81033d25119fdf5f8313cedeaa88a7932/sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubProducerAsyncClient.java#L516-L522

In my code I have tried calling it with reflection and this does support the behaviour I'm looking for:

```java
    eventHubProducerAsyncClient.getPartitionIds()
            .switchIfEmpty(Mono.error(new IllegalStateException("No partition for event hub: " + eventHubName)))
            .flatMap(eventHubProducerAsyncClient::getSendLink) // call to private method
            .flatMap(AmqpLink::getEndpointStates)
            .filter(amqpEndpointState -> amqpEndpointState == AmqpEndpointState.ACTIVE)
            .switchIfEmpty(Mono.error(new IllegalStateException("No active link found for event hub: " + eventHubName)))
            .blockFirst(Duration.ofSeconds(healthCheckTimeoutSeconds));
```

So I would like to see this method become part of the public API, if not directly then with a wrapper that could encapsulate the above behaviour.

**Describe alternatives you've considered**
I also considered that it might be nice to expose the Status of the Event Hub (e.g. Active, Disabled, SendDisabled), something which I couldn't find today.

I also wondered about being able to validate the permissions for a token (e.g. the one in the connection string) without actually trying to use it to authenticate a call, but I'm not sure if this is possible.

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Description Added
- [x] Expected solution specified

**Comment by joshfree** [comment]
@conniey could you follow up with @scootafew please?

**Comment by conniey** [comment]
@scootafew We were intentional about lazily initialising the connection/send link. You can perform an operation like getEventHubInformation() which will create a connection to the service and get information about the Event Hub without sending data.

/cc @ramya-rao-a Do you know if we had any health check in our backlog?

**Comment by scootafew** [comment]
> You can perform an operation like getEventHubInformation() which will create a connection to the service and get information about the Event Hub without sending data.

@conniey I see `getEventHubProperties()`, but that returns surprisingly little information; just the name, creation time and the ability to list the partition IDs. From what I can tell it contacts the Event Hub's management endpoint, which tells me the Event Hub exists and is reachable, but not if it's disabled or whether my connection string has the send permission.

**Comment by conniey** [comment]
@JamesBirdsall Do you know of a way we can check that a connection string has send permissions without actually sending a message?

**Comment by conniey** [comment]
@scootafew if you invoke `createBatch()`, that will create an AMQP send link and authorize with the CBS node. So if you don't have authorization to send, it'll end in an error.

**Comment by JamesBirdsall** [comment]
This is a complicated question with a lot of tradeoffs.
1) The expectation was that applications looking for most of the information @scootafew mentions would do so via the management API, not runtime -- control plane, not data plane -- BUT that requires management permission, which a simple producer application would not otherwise need.
2) The service only gets the specific event hub name when a link is created, so at runtime things like auth, enabled/disabled, or even existence can't be checked until and unless a link is created. createBatch() creates a link in order to get the maximum batch size and should throw if sending is not allowed for whatever reason. But creating the link is an implementation detail, not contractual, so it could change in the future if we develop a different way for the client to get the max batch size.
3) Permissions, enabled/disabled, existence, etc. can all change at any time while the producer app is running. I can understand the desire to fail fast if the config is bad at startup, but the config can become bad at any time and the application and monitoring needs to be ready to handle and detect that.

Bottom line: createBatch() seems like a workaround that should deliver most or all of what you want right now and will probably continue to do so, or at least not cause false alarms in the future if things change, but there's a larger picture that is worth considering too.

**Comment by conniey** [comment]
> But I now have another question @conniey, as you say this only works on the first call; does this mean that an open AMQP send link can continue to be used even if the credentials used to create it have been revoked? Because I think I have seen this behaviour, although I wasn't sure, as I would have expected that at least an error would be thrown when trying to send. Or is it just that it can take a period of time to propagate the change?

We authorize with the CBS periodically, passing in an access token with a duration of 20 minutes to ensure that the sender is authorized with that credential.
